kartothek/io/testing/update.py

```diff
@@ -673,7 +680,8 @@ def test_partition_on_null(store_factory, bound_update_dataset):  # gh-262
     )
 
     with pytest.raises(
-        ValueError, match=r"Original dataframe size .* on a column with null values."
```

Comment:
I'd also be fine with just having `Exception` here instead of `ValueError`. @usha-nemani-jdas would that fix your issue?

Reply:
Exception will fix my issue, since it is parent of ValueError and YamalError . It would support all other Error Types as well. Will do code changes accordingly.

Reply:
I am not comfortable with the idea of generalizing it to an `Exception` when we know what type of exception to expect...

Reply:
We do not have a proper custom exception for this error case anyway, which imo would be the most correct way of handling it. I'm fine with the tradeoff for test simplicity in this case as we are also matching the error string

Reply:
@KrishnarajAjayGharpure are you okay if we merge this?

Reply:
@lr4d - you make a good point about the string matching. However, I think this change is small enough to not overly worry and merge in the interest of expediency.